### PR TITLE
User Syncronization

### PR DESCRIPTION
**Pull request by MadsLudvig**

### CODE REVIEW

File: src/boards.ts [hunk 1]
```diff
@@ -1,150 +1,208 @@
...
     this.namespace.on("connection", (socket) => {
       socket.on("startDraw", this.handleStartDraw.bind(this));
       socket.on("doDraw", this.handleDoDraw.bind(this));
       socket.on("startErase", this.handleStartErase.bind(this));
       socket.on("doErase", this.handleDoErase.bind(this));
+      socket.on("startMove", this.handleStartMove.bind(this));
+      socket.on("doMove", this.handleDoMove.bind(this));
       socket.on("undo", this.handleUndo.bind(this));
       socket.on("redo", this.handleRedo.bind(this));
+      socket.on("disconnect", () => {
```
Comment (tbdlarsen) [inline src/boards.ts:97]: How do you save offline users ?

File: src/commandController.ts [hunk 2]
```diff
@@ -1,64 +1,66 @@
...
+    this.stack = new Map();
     this.namespace = namespace;
   }
-  execute(command: DrawCommand | MoveCommand | EraseCommand, username: string) {
+
+  /**
+   * Executes a command, adding in to the stack. Removes all undone elements belonging to user
+   * @param command - Command to be executed
+   * @param username - user which executes the command
+   */
```
Comment (tbdlarsen) [inline src/commandController.ts]: undo redo af draw og move ????

File: src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { CoordinateSet } from "./types";
```
Comment (tbdlarsen) [inline src/utils.ts]: Where is CoordinateSet defined

**Comment by MadsLudvig** [comment]
This branch needs testing with user-sync branch on Magicboard repo

File: src/commandController.ts [hunk 4]
```diff
@@ -1,64 +1,68 @@
...
+  execute(
+    command: DrawCommand | EraseCommand | MoveCommand,
```
Comment (krestenlaust) [inline src/commandController.ts]: 👀

**Review by tbdlarsen — COMMENTED** [review]
tilstraekkeligt

**Review by krestenlaust — APPROVED** [review]

**Pull request merged**